.travis.yml

```diff
@@ -1,52 +1,42 @@
-########################################################################
-# This pipeline is executed on PHP 7.4:
-# - required for development
-# - required to run Rector, to generate the PHP 7.1-compatible code for production
-#
-# It uses the Build Environment "Xenial", because it has PHP 7.1 preinstalled,
-# so we can switch from the global PHP 7.4 to local PHP 7.1 through `phpenv`
-#
-# There are 2 parallel executions, with env variable:
-# 1. DOWNGRADE_CODE="true"
-# 2. DOWNGRADE_CODE="false"
-#
-# When DOWNGRADE_CODE is "false", it is the standard testing pipeline:
-# Execute PHPCS, PHPUnit and PHPStan to check there are no issues, and generate report.
-#
-# When DOWNGRADE_CODE is "true", it tests the conversion of the PHP code version 7.4 to 7.1:
-# - Run Rector to downgrade the code
-# - Run PHPStan. If it exits with no errors, then downgrading the code works well.
-########################################################################
-dist: xenial
+# TravisCI configuration for GraphQLAPI/graphql-api-for-wp
 language: php
+os:
+  - linux
+dist: bionic
+
+cache:
+  directories:
+    - "${HOME}/.composer/cache"
+
+php:
+  - 7.4
 
 jobs:
   include:
-  - php: 7.4
-    env: DOWNGRADE_CODE=false
-  - php: 7.4
-    env: DOWNGRADE_CODE=true
+    - name: "PHPCS"
+      script:
+        - vendor/bin/phpcs -n --standard=PSR2 src/
+      after_script: skip
 
-## Cache composer
-cache:
-  directories:
-    - $HOME/.composer/cache
+    - name: "PHPStan"
+      script:
+        - vendor/bin/phpstan analyse src/
+      after_script: skip
+
+    - name: "PHP 7.1"
+      install:
+        - travis_retry composer update --no-interaction --prefer-dist
+        - vendor/bin/rector process
+        - phpenv local 7.1
+        - php --version
+        - vendor/bin/phpstan analyse src/
+      after_script: skip
 
-before_script:
-    - travis_retry composer update ${COMPOSER_FLAGS} --no-interaction --prefer-dist
-    # Downgrade the code to PHP 7.1 via Rector
-    - if [[ "$DOWNGRADE_CODE" == "true" ]]; then vendor/bin/rector process; fi
+install:
+  - travis_retry composer update --no-interaction --prefer-dist
 
 script:
-  # Tests only on PHP 7.4
-  - if [[ "$DOWNGRADE_CODE" != "true" ]]; then vendor/bin/phpcs -n --standard=psr2 src/; fi
-  - if [[ "$DOWNGRADE_CODE" != "true" ]]; then vendor/bin/phpunit --coverage-text --coverage-clover=coverage.clover; fi
-  # Switch to PHP 7.1 to test if downgraded code works, then up again to 7.4
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then phpenv local 7.1; fi
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then php --version; fi
-  - vendor/bin/phpstan analyse -c phpstan.neon.dist src
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then phpenv local 7.4; fi
-  - if [[ "$DOWNGRADE_CODE" == "true" ]]; then php --version; fi
+  - vendor/bin/phpunit --coverage-text --coverage-clover=coverage.clover
```

Comment:
This must not run on the PHP 7.1 job. It will always fail, because I'm using PHPUnit 9.3

Reply:
This does not run on PHP 7.1
There are 1+3 jobs.

Reply:
It did run, that's why it failed

Reply:
Now it's working :)